src/vaults/factories/PoolEscrowFactory.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {Auth} from "src/misc/Auth.sol";
+
+import {
+    IPoolEscrowProvider,
+    IEscrowProvider,
+    IPoolEscrowFactory
+} from "src/vaults/interfaces/factories/IPoolEscrowFactory.sol";
+import {IPoolEscrow} from "src/vaults/interfaces/IEscrow.sol";
+import {PoolEscrow} from "src/vaults/Escrow.sol";
+
+contract PoolEscrowFactory is IPoolEscrowFactory, Auth {
+    uint64 public constant V2_POOL_ID = 4139607887;
+    address public constant V2_GLOBAL_ESCROW = address(0x0000000005F458Fd6ba9EEb5f365D83b7dA913dD);
+
+    address public immutable root;
+    address public poolManager;
+    address public balanceSheet;
+    address public asyncRequests;
+
+    mapping(uint64 poolId => address) public escrows;
+
+    constructor(address root_, address deployer) Auth(deployer) {
+        root = root_;
+    }
+
+    /// @inheritdoc IPoolEscrowFactory
+    function file(bytes32 what, address data) external auth {
+        if (what == "poolManager") poolManager = data;
+        else if (what == "balanceSheet") balanceSheet = data;
+        else if (what == "asyncRequests") asyncRequests = data;
+        else revert FileUnrecognizedParam();
+        emit File(what, data);
+    }
+
+    /// @inheritdoc IPoolEscrowFactory
+    function newEscrow(uint64 poolId) public auth returns (address) {
```

Comment:
NIT. Could we return here a `IPoolEscrow`?

Reply:
Due to the refactor in [d72ad0a](https://github.com/centrifuge/protocol-v3/pull/262/commits/d72ad0a1e9fe0998e58731d1f0ea155a38fc0835) which removed `poolEscrow` and `IEscrowProvider`, I decided to just return the address and require consumers to type cast if necessary.

Reply:
Ok. Just to notify, if you return an address instead of `IPoolEscrow` to avoid breaking retro-compatibility, both types, AFAIK, have the same abi. 

Is there any reason to keep the `address` as type when it always is a `IPoolEscrow` interface?

Reply:
[99cb82a](https://github.com/centrifuge/protocol-v3/pull/262/commits/99cb82a42bf8505c80782b08cfcd664d803b9fd6)